sapl/templates/sessao/adicionar_varias_materias_expediente.html

```diff
@@ -126,5 +143,22 @@ <h3>{% trans 'Pesquisa concluída com sucesso! Foi encontrada 1 matéria.'%}</h3
 		$(window).on('beforeunload', function() {
       $("input[type=submit], input[type=button]").prop("disabled", "disabled")
     });
+    function checkAll(elem) {
+      let checkboxes = document.getElementsByName('materia_id');
+      for (let i = 0; i < checkboxes.length; i++) {
+        if (checkboxes[i].type == 'checkbox')
+            checkboxes[i].checked = elem.checked;
+      }
+    }
+
+    function marcaTipoVotacao() {
+      materias = document.getElementsByName('materia_id');
+      for (var i=0; i<materias.length;i++){
+        if ( materias[i].checked ){
+           var id_materia = "#tipo_votacao_" + materias[i].value + "_" + document.querySelector('input[name="tipo_votacao_check_all"]:checked').value;
+           $(id_materia).prop('checked', true);
+        }
+      }
```

Comment:
Mesmo comentário acima, utilizar jQuery para deixar código mais conciso e legível.

```suggestion
$('input[name="materia_id"]:checked').each(function() {
  var tipoVotacao = $('input[name="tipo_votacao_check_all"]:checked').val();
  var idMateria = "#tipo_votacao_" + $(this).val() + "_" + tipoVotacao;
  $(idMateria).prop('checked', true);
});
```